converter/internal/prometheusconvert/prometheusconvert.go

```diff
@@ -51,7 +61,32 @@ func Convert(in []byte) ([]byte, error) {
 
 	var buf bytes.Buffer
 	if _, err := f.WriteTo(&buf); err != nil {
-		return nil, fmt.Errorf("failed to render Flow config: %w", err)
+		diags.Add(diag.Diagnostic{
+			Severity: diag.SeverityLevelError,
+			Message:  fmt.Sprintf("failed to render Flow config: %s", err.Error()),
+		})
+		return nil, diags
+	}
+	return buf.Bytes(), diags
+}
+
+// ValidateUnsupported will traverse the Prometheus Config and return warnings
+// for any config we knowingly do not support.
+func ValidateUnsupported(promConfig *promconfig.Config) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	for _, scrapeConfig := range promConfig.ScrapeConfigs {
+		for _, sdConfig := range scrapeConfig.ServiceDiscoveryConfigs {
+			switch sdConfig.(type) {
+			case *nomad.SDConfig:
+				diags.Add(diag.Diagnostic{
+					Severity: diag.SeverityLevelWarn,
+					Message:  "unsupported nomad_sd_config was provided",
+				})
+			default:
+			}
```

Comment:
I know this is just a prototype, but protip:


```suggestion
			switch sdConfig.(type) {
			default:
				diags.Add(diag.Diagnostic{
					Severity: diag.SeverityLevelWarn,
					Message:  fmt.Sprintf("unsupported %s was provided", sdConfig.Name()),
				})
			}
```

Reply:
I think I may need more granular decision making here since we do implement a number of discovery components and I think this will warn on all of them as suggested. I could do it this way as long as I can use the sdConfig.Name() to differentiate ones we have from ones we don't or leave as is.

Reply:
Sorry, I was noting that you could use `default` for warning against any discovery which isn't supported, and add specific cases for ones that are :)

Reply:
ah yes reverse the logic and auto catch misses. I like that